orca-core/src/main/groovy/com/netflix/spinnaker/orca/listeners/StageStatusPropagationListener.groovy

```diff
@@ -43,7 +43,12 @@ class StageStatusPropagationListener implements StageListener {
     if (executionStatus) {
       def nonBookendTasks = stage.tasks.findAll { !it.bookend }
       def lastTask = task.stageEnd || task.name == "stageEnd"
-      // TODO: 2nd part is v1 engine — to be removed
+      if (stage.syntheticStageOwner && task == nonBookendTasks[-1]) {
```

Comment:
v1-style executions do not have startStage/endStage markers for synthetic stages.